### ISSUE TITLE
Distance issues with Kinect and R Pi 4

### ISSUE THREAD

**Issue by piran-prog** [OpenKinect/libfreenect#662]
Along with the other issues I'm having I've been trying the regview and I'm getting things like this:
![image](https://user-images.githubusercontent.com/75266063/189542290-b7b16ff0-8a05-4d9b-8a51-151a32749cab.png)
The foreground is lighter and then shades to dark blue over about 2 meteres and then just shows the RGB image but then further away starts showing distance colours again

**Comment by piedar** [comment]
I think this is a limitation of the depth shading algorithm in `freenect-regview` but I'm not sure how to fix it.  Ideally there should be a smooth color range from near to far.  As far as I can tell, the actual depth values are correct.

**Comment by piran-prog** [comment]
Yes looks like an issue with regview, some of the other test apps seem to get it right. I managed (with a lot of effort) to get it working in ROS and that works but I'm trying to get it working with ORBSLAM3 and it's getting some of the depth wrong there: 
![image](https://user-images.githubusercontent.com/75266063/191552828-19ba0d3f-ce5c-4aa3-b5d2-2dfda453c4bb.png)
where the upper cloud of points is all on a single plane but should actually be at the front with the other points.
